soccer/gameplay/role_assignment.py

```diff
@@ -181,11 +181,13 @@ def assign_roles(robots, role_reqs):
     # To do our calculations though, we need to have a flat list of RoleRequirements to match to our list of robots
     # The flatten_tree() method does this and keeps track of how to rebuild the tree so we can do so at the end
     def flatten_tree(tree, path_prefix=[]):
+        valid_shell_ids = [bot.shell_id() for bot in robots]
         for key, subtree in tree.items():
             if isinstance(subtree, dict):
                 flatten_tree(subtree, path_prefix + [key])
             elif isinstance(subtree, RoleRequirements):
-                role_reqs_list.append(subtree)
+                if subtree.required_shell_id == None or subtree.required_shell_id in valid_shell_ids:
```

Comment:
@jgkamat I will go ahead and change this. One thing I wanted to point out is that I noticed that the checks in the [setup_goalie_if_needed](https://github.com/RoboJackets/robocup-software/blob/209e1105b98de70713596a2f9a0054c0497bda8f/soccer/gameplay/root_play.py#L199) function of root_play use `== None`, I'm not sure if that really matters or not though.